Closes #19751: Disable occupied Module Bays in form dropdowns

pheus opened the pull request:
### Fixes: #19751

Implements the occupied‑bay UI behavior without adding a private `_occupied` field to the API.

Update the `module_bay` form field to use context with `disabled_indicator: 'installed_module'` so occupied bays are disabled in dropdowns

**Notes**
- Backward compatible; no database or migration changes
- Keeps the REST surface clean while providing the needed UI parity

arthanson reviewed (changes requested):
This is showing up in the regular REST API (see screenshot) which with an underscore seems like it is an internal / private parameter? 
<img width="610" height="992" alt="Monosnap Image 2025-10-17 14-25-32" src="https://github.com/user-attachments/assets/ea2268a1-701b-4989-ba61-d848c6486c5e" />

pheus commented:
Thanks for the thoughtful review!

You’re right! The leading underscore makes it look internal. I initially mirrored `CabledObjectSerializer` (which exposes `_occupied`), but I agree it’s not ideal here and a brief‑only approach would still require listing it in `Meta.fields`.

I’ve reworked this to keep the API surface clean:

- Drop `_occupied` entirely
- Use `installed_module` as the occupancy signal (included in brief)
- Update the form field context to disable options via `installed_module`

This preserves the UI behavior without adding a private‑looking field. If you’d prefer an explicit boolean, I can expose a read‑only `occupied` derived from `installed_module`. Happy to adjust. Thanks again!

pheus committed:
feat(forms): Add context handling for ModuleBay field

Extend the context of the `module_bay` field to support a `disabled`
parameter. This facilitates the handling of installed modules and adds
flexibility to form field rendering.

Fixes #19751

pheus commented:
Apologies for the noise! I tried to rename the feature branch in GitHub’s UI to better reflect the updated scope, and that inadvertently auto‑closed this PR. I’ve restored the branch and reopened the PR—no code changes were introduced as part of this. Sorry for the disruption!

arthanson reviewed (approved):
Thanks @pheus! much cleaner implementation.
